In [2]:
import sys, os

# go two levels up:  .../Graphnet-Applications
repo_root = os.path.abspath("../..")
sys.path.append(repo_root)
from Helpers.tabulators import FrameKeyToTable


In [3]:
from icecube import dataio, icetray

In [4]:
import pandas as pd


In [5]:
from icecube import LeptonInjector # for EventProperties


In [6]:
DATA_PATH = "/project/def-nahee/kbas/POM_Response/pom_response_batch_000.i3.gz"
GCD_PATH = "/project/6008051/pone_simulation/GCD_Library/PONE_800mGrid.i3.gz"

## here, frame corresponds to one event. All Keys in the `Frame` are:

In [7]:
data_file = dataio.I3File(DATA_PATH)
data_file.pop_frame()
frame = data_file.pop_frame()
print(frame)

[ I3Frame  (Physics):
  'EventProperties' [DAQ] ==> LeptonInjector::BasicEventProperties (140)
  'EventPulseSeries' [DAQ] ==> I3Map<OMKey, vector<I3RecoPulse> > (167406)
  'I3EventHeader' [DAQ] ==> I3EventHeader (99)
  'I3MCTree' [DAQ] ==> TreeBase::Tree<I3Particle, I3ParticleID, i3hash<I3ParticleID> > (422)
  'I3MCTree_RNGState' [DAQ] ==> I3GSLRandomServiceState (87)
  'I3MCTree_postprop' [DAQ] ==> TreeBase::Tree<I3Particle, I3ParticleID, i3hash<I3ParticleID> > (2778)
  'MMCTrackList' [DAQ] ==> I3Vector<I3MMCTrack> (304)
  'Noise_Dark' [DAQ] ==> I3Map<OMKey, vector<I3RecoPulse> > (52540)
  'Noise_K40' [DAQ] ==> I3Map<OMKey, vector<I3RecoPulse> > (165492)
  'PMT_Response' [DAQ] ==> I3Map<OMKey, vector<I3RecoPulse> > (222554)
  'PMT_Response_nonoise' [DAQ] ==> I3Map<OMKey, vector<I3RecoPulse> > (1337)
  'TriggerTime_3PMT_1DOM' [DAQ] ==> I3PODHolder<double> (36)
]



# Let's not look at all these keys individually in 3 sections:
- ## 1: Event Based Keys
    - EventProperties
    - I3EventHeader
- ## 2: EventPulseSeries, PMT_Response, PMT_Response_nonoise
- ## 3: I3MCTree and I3MCTree_postprop, MMCTrackList, 
- ## 4: Noises
    - Noise_K40
    - Noise_Dark
- ## 5: Not Useful Metadata
    - I3MCTree_RNGState (check its meaning)

TriggerTime_3PMT_1DOM?

# 1.1: EventProperties

In [8]:
ep_table = FrameKeyToTable(
    data_path=DATA_PATH,
    frame_key="EventProperties",
)
df_ep = ep_table.to_dataframe(max_events=300)
df_ep

azimuth  finalStateX  finalStateY  finalType1  finalType2  \
0   1.378987     0.268484     0.184845          13 -2000001006   
1   1.610513     0.144349     0.068259          13 -2000001006   
2   2.099105     0.069946     0.459941          13 -2000001006   
3   5.724277     0.129059     0.251009          13 -2000001006   
4   3.115837     0.010576     0.489280          13 -2000001006   
5   4.404895     0.259356     0.820747          13 -2000001006   
6   3.129628     0.086363     0.284626          13 -2000001006   
7   3.387027     0.114479     0.168782          13 -2000001006   
8   0.798343     0.400808     0.369215          13 -2000001006   
9   3.509205     0.148686     0.777909          13 -2000001006   
10  1.355114     0.160379     0.821264          13 -2000001006   
11  3.158192     0.046245     0.422460          13 -2000001006   
12  4.438802     0.123975     0.371050          13 -2000001006   
13  1.900578     0.009076     0.144985          13 -2000001006   
14  1.492162     0.024774     0.808911          13 -2000001006   
15  3.505507     0.632790     0.693376          13 -2000001006   
16  2.286771     0.280561     0.687120          13 -2000001006   
17  0.835354     0.490356     0.188121          13 -2000001006   
18  4.443948     0.080454     0.310663          13 -2000001006   
19  3.152562     0.159003     0.168131          13 -2000001006   
20  4.863213     0.358015     0.174003          13 -2000001006   
21  2.404312     0.262949     0.362201          13 -2000001006   
22  1.372047     0.149641     0.968859          13 -2000001006   
23  5.241702     0.486630     0.536193          13 -2000001006   
24  1.395472     0.016496     0.444130          13 -2000001006   
25  0.874603     0.412811     0.808901          13 -2000001006   
26  1.087445     0.093002     0.008794          13 -2000001006   
27  0.924961     0.283693     0.670714          13 -2000001006   
28  2.203876     0.121852     0.428276          13 -2000001006   
29  1.892135     0.001602     0.867434          13 -2000001006   
30  0.190961     0.215733     0.459027          13 -2000001006   
31  0.063743     0.263793     0.973128          13 -2000001006   
32  4.262830     0.159411     0.427433          13 -2000001006   
33  1.748562     0.005963     0.511907          13 -2000001006   
34  5.161979     0.512081     0.023166         -13 -2000001006   
35  2.344937     0.100273     0.145503         -13 -2000001006   
36  5.523766     0.277673     0.913720         -13 -2000001006   
37  5.272596     0.009753     0.271066         -13 -2000001006   
38  1.137987     0.315962     0.288141         -13 -2000001006   
39  4.189748     0.245291     0.010757         -13 -2000001006   
40  3.914233     0.043672     0.271836         -13 -2000001006   
41  3.766810     0.211394     0.487680         -13 -2000001006   
42  2.509762     0.196378     0.636522         -13 -2000001006   
43  6.121448     0.320786     0.743096         -13 -2000001006   
44  2.788258     0.161165     0.529215         -13 -2000001006   
45  2.256732     0.105372     0.138791         -13 -2000001006   
46  4.477656     0.370201     0.103189         -13 -2000001006   
47  4.599359     0.062872     0.925328         -13 -2000001006   
48  3.256038     0.006773     0.154589         -13 -2000001006   
49  3.705823     0.180953     0.617008         -13 -2000001006   
50  0.418770     0.126383     0.276465         -13 -2000001006   
51  3.051141     0.082848     0.063945         -13 -2000001006   
52  0.539853     0.123816     0.143805         -13 -2000001006   
53  0.582255     0.353224     0.013921         -13 -2000001006   
54  4.060421     0.191928     0.163324         -13 -2000001006   
55  1.645004     0.079905     0.498043         -13 -2000001006   
56  0.933296     0.034198     0.528592         -13 -2000001006   

    impactParameter  initialType       radius  totalColumnDepth  \
0        418.869970           14   347.765925      2.825059e+05   
1        585.847090           14   568.444

### FinalStateY 

`FinalStateY` is the **inelasticity** (often called Bjorken-y) of the neutrino interaction: It tells you **what fraction of the neutrino energy goes into the hadronic system**.
#### Plugging in the numbers for the first event  (buraya baska notlar da eklemelisin)

- `TotalEnergy = NuMu energy`  = 119.301 GeV
- `Energy of Hadrons`= 22.0522 GeV
- `FinalStateY`:  22.0522 GeV/119.301 GeV = 0.184845 

So, in the first event:

> about **18.5%** of the neutrino energy went into the **hadronic cascade**,  
> and about **81.5%** went into the **muon track**.


### FinalStateX (I did not check this myself)

`FinalStateX (Bjorken-x)`:

$$
x = \frac{Q^2}{2\,m_N\,(E_\nu - E_\mu)}
$$

and using inelasticity, 

$$
x = \frac{Q^2}{2\,m_N\,E_\nu\,y}
$$

Where:

- $x$: Bjorken-x (FinalStateX)
- $Q^2$: squared four-momentum transfer ($Q^2 = -q^2$)
- $m_N$: nucleon mass (proton or neutron)
- $E_\nu$: incoming neutrino energy
- $E_\mu$: outgoing muon energy
- $y$: inelasticity (FinalStateY), approximately $y \approx (E_\nu - E_\mu)/E_\nu$

### TotalColumnDepth 
### Radius 
### ImpactParameter 



# 1.2: I3EventHeader

In [9]:
i3eh_table = FrameKeyToTable(
    data_path=DATA_PATH,
    frame_key="I3EventHeader",
)
df_i3eh = i3eh_table.to_dataframe(max_events=300)
df_i3eh

CONFIG_IN_TRANSITION  OK  UNKNOWN_STATE data_stream  \
0                     40  20              0     Physics   
1                     40  20              0     Physics   
2                     40  20              0     Physics   
3                     40  20              0     Physics   
4                     40  20              0     Physics   
5                     40  20              0     Physics   
6                     40  20              0     Physics   
7                     40  20              0     Physics   
8                     40  20              0     Physics   
9                     40  20              0     Physics   
10                    40  20              0     Physics   
11                    40  20              0     Physics   
12                    40  20              0     Physics   
13                    40  20              0     Physics   
14                    40  20              0     Physics   
15                    40  20              0     Physics   
16                    40  20              0     Physics   
17                    40  20              0     Physics   
18                    40  20              0     Physics   
19                    40  20              0     Physics   
20                    40  20              0     Physics   
21                    40  20              0     Physics   
22                    40  20              0     Physics   
23                    40  20              0     Physics   
24                    40  20              0     Physics   
25                    40  20              0     Physics   
26                    40  20              0     Physics   
27                    40  20              0     Physics   
28                    40  20              0     Physics   
29                    40  20              0     Physics   
30                    40  20              0     Physics   
31                    40  20              0     Physics   
32                    40  20              0     Physics   
33                    40  20              0     Physics   
34                    40  20              0     Physics   
35                    40  20              0     Physics   
36                    40  20              0     Physics   
37                    40  20              0     Physics   
38                    40  20              0     Physics   
39                    40  20              0     Physics   
40                    40  20              0     Physics   
41                    40  20              0     Physics   
42                    40  20              0     Physics   
43                    40  20              0     Physics   
44                    40  20              0     Physics   
45                    40  20              0     Physics   
46                    40  20              0     Physics   
47                    40  20              0     Physics   
48                    40  20              0     Physics   
49                    40  20              0     Physics   
50                    40  20              0     Physics   
51                    40  20              0     Physics   
52                    40  20              0     Physics   
53                    40  20              0     Physics   
54                    40  20              0     Physics   
55                    40  20              0     Physics   
56                    40  20              0     Physics   

                                 end_time  event_id  run_id  \
0   1905-06-04 18:35:44.000,000,000,0 UTC         1       0   
1   1905-06-04 18:35:44.000,000,000,0 UTC        11       0   
2   1905-06-04 18:35:44.000,000,000,0 UTC        14       0   
3   1905-06-04 18:35:44.000,000,000,0 UTC        15       0   
4   1905-06-04 18:35:44.000,000,000,0 UTC        18       0   
5   1905-06-04 18:35:44.000,000,000,0 UTC        20       0   
6   1905-06-04 18:35:44.000,000,000,0 UTC        22       0   
7   1905-06-04 18:35:44.000,000,000,0 UTC        29       0   
8   1905-06-04

###### some observations etc related to the table here

# 2.1: EventPulseSeries


In [10]:
eps_table = FrameKeyToTable(
    data_path=DATA_PATH,
    frame_key="EventPulseSeries",     # I3Map<OMKey, vector<I3RecoPulse>>
)

df_eps = eps_table.to_dataframe(max_events=300)
df_eps

min_string_id  max_string_id  min_om_id  max_om_id  min_pmt_id  \
0             1.0          340.0        1.0       20.0         1.0   
1             1.0          340.0        1.0       20.0         1.0   
2             1.0          340.0        1.0       20.0         1.0   
3             1.0          340.0        1.0       20.0         1.0   
4             1.0          340.0        1.0       20.0         1.0   
5             1.0          340.0        1.0       20.0         1.0   
6             1.0          340.0        1.0       20.0         1.0   
7             1.0          340.0        1.0       20.0         1.0   
8             1.0          340.0        1.0       20.0         1.0   
9             1.0          340.0        1.0       20.0         1.0   
10            1.0          340.0        1.0       20.0         1.0   
11            1.0          340.0        1.0       20.0         1.0   
12            1.0          340.0        1.0       20.0         1.0   
13            1.0          340.0        1.0       20.0         1.0   
14            1.0          340.0        1.0       20.0         1.0   
15            1.0          340.0        1.0       20.0         1.0   
16            1.0          340.0        1.0       20.0         1.0   
17            1.0          340.0        1.0       20.0         1.0   
18            1.0          340.0        1.0       20.0         1.0   
19            1.0          340.0        1.0       20.0         1.0   
20            1.0          340.0        1.0       20.0         1.0   
21            1.0          340.0        1.0       20.0         1.0   
22            1.0          340.0        1.0       20.0         1.0   
23            1.0          340.0        1.0       20.0         1.0   
24            1.0          340.0        1.0       20.0         1.0   
25            1.0          340.0        1.0       20.0         1.0   
26            NaN            NaN        NaN        NaN         NaN   
27            1.0          340.0        1.0       20.0         1.0   
28            1.0          340.0        1.0       20.0         1.0   
29            1.0          340.0        1.0       20.0         1.0   
30            1.0          340.0        1.0       20.0         1.0   
31            1.0          340.0        1.0       20.0         1.0   
32            1.0          340.0        1.0       20.0         1.0   
33            1.0          340.0        1.0       20.0         1.0   
34            1.0          340.0        1.0       20.0         1.0   
35            1.0          340.0        1.0       20.0         1.0   
36            1.0          340.0        1.0       20.0         1.0   
37            1.0          340.0        1.0       20.0         1.0   
38            1.0          340.0        1.0       20.0         1.0   
39            1.0          340.0        1.0       20.0         1.0   
40            1.0          340.0        1.0       20.0         1.0   
41            1.0          340.0        1.0       20.0         1.0   
42            1.0          340.0        1.0       20.0         1.0   
43            1.0          340.0        1.0       20.0         1.0   
44            1.0          340.0        1.0       20.0         1.0   
45            1.0          340.0        1.0       20.0         1.0   
46            1.0          340.0        1.0       20.0         1.0   
47            1.0          340.0        1.0       20.0         1.0   
48            1.0          340.0        1.0       20.0         1.0   
49            1.0          340.0        1.0       20.0         1.0   
50            1.0          340.0        1.0       20.0         1.0   
51            1.0          340.0        1.0       20.0         1.0   
52            1.0          340.0        1.0       20.0         1.0   
53            1.0          340.0        1.0       20.0         1.0   
54            1.0          340.0        1.0       20.0         1.0   
55            1.0          340.0        1.0       20.0         1.0   
56            

In [11]:
# data_file = dataio.I3File(DATA_PATH)
# data_file.pop_frame()
# frame = data_file.pop_frame()
# print(frame["EventPulseSeries"])

# 2.2: PMT_Response

In [12]:
pmtr_table = FrameKeyToTable(
    data_path=DATA_PATH,
    frame_key="PMT_Response",     # I3Map<OMKey, vector<I3RecoPulse>>
)

df_pmtr = pmtr_table.to_dataframe(max_events=300)
df_pmtr

min_string_id  max_string_id  min_om_id  max_om_id  min_pmt_id  \
0               1            340          1         20           1   
1               1            340          1         20           1   
2               1            340          1         20           1   
3               1            340          1         20           1   
4               1            340          1         20           1   
5               1            340          1         20           1   
6               1            340          1         20           1   
7               1            340          1         20           1   
8               1            340          1         20           1   
9               1            340          1         20           1   
10              1            340          1         20           1   
11              1            340          1         20           1   
12              1            340          1         20           1   
13              1            340          1         20           1   
14              1            340          1         20           1   
15              1            340          1         20           1   
16              1            340          1         20           1   
17              1            340          1         20           1   
18              1            340          1         20           1   
19              1            340          1         20           1   
20              1            340          1         20           1   
21              1            340          1         20           1   
22              1            340          1         20           1   
23              1            340          1         20           1   
24              1            340          1         20           1   
25              1            340          1         20           1   
26              1            340          1         20           1   
27              1            340          1         20           1   
28              1            340          1         20           1   
29              1            340          1         20           1   
30              1            340          1         20           1   
31              1            340          1         20           1   
32              1            340          1         20           1   
33              1            340          1         20           1   
34              1            340          1         20           1   
35              1            340          1         20           1   
36              1            340          1         20           1   
37              1            340          1         20           1   
38              1            340          1         20           1   
39              1            340          1         20           1   
40              1            340          1         20           1   
41              1            340          1         20           1   
42              1            340          1         20           1   
43              1            340          1         20           1   
44              1            340          1         20           1   
45              1            340          1         20           1   
46              1            340          1         20           1   
47              1            340          1         20           1   
48              1            340          1         20           1   
49              1            340          1         20           1   
50              1            340          1         20           1   
51              1            340          1         20           1   
52              1            340          1         20           1   
53              1            340          1         20           1   
54              1            340          1         20           1   
55              1            340          1         20           1   
56            

In [13]:
# data_file = dataio.I3File(DATA_PATH)
# data_file.pop_frame()
# frame = data_file.pop_frame()
# print(frame["PMT_Response"])

# 2.3: PMT_Response_nonoise

In [14]:
pmtrnn_table = FrameKeyToTable(
    data_path=DATA_PATH,
    frame_key="PMT_Response_nonoise",     # I3Map<OMKey, vector<I3RecoPulse>>
)

df_pmtrnn = pmtrnn_table.to_dataframe(max_events=300)
df_pmtrnn

min_string_id  max_string_id  min_om_id  max_om_id  min_pmt_id  \
0             180            256         16         20           1   
1             148            187         18         20           5   
2               7             19          4          6           1   
3               8             46         15         20           1   
4               4             69          1         20           1   
5             332            340         19         20           1   
6               7             23          3          7           1   
7             264            332          7         20           1   
8              15             59          6         10           1   
9             297            324         16         18           1   
10            151            264          7         14           1   
11            204            241         18         20           1   
12              6            119          3          7           1   
13             28            258          4         17           1   
14             13             39          8         10           1   
15             38            110          1         13           1   
16            293            320         16         17           1   
17             30             82          1          5           1   
18             53            203          9         20           1   
19             36             73          1          5           1   
20             15            240         12         15           1   
21            140            179         18         20           1   
22             13             38         17         20           1   
23            119            215          9         19           1   
24              4            126          9         20           1   
25            246            263          1          7           1   
26            225            281          1          2           6   
27            225            261         18         20           1   
28             94            291         17         20           1   
29            244            333         14         16           1   
30            132            154         13         17           1   
31             82            101          1          2           6   
32             15             93          1         13           1   
33              2            336          1         17           1   
34             44             95          1          2           1   
35             13            159         11         20           1   
36             25             55          2          5           1   
37            263            323         12         16           1   
38            306            320         17         20           2   
39            223            310          1         19           1   
40              8             35          4          6           1   
41             54             88          1          3           1   
42              7             60         12         17           1   
43             48             82          4          7           1   
44            128            206          2         18           1   
45            252            312         16         18           2   
46            124            276          1         12           1   
47            164            184         15         17           1   
48             77            105          1         15           1   
49             95            134          3          6           1   
50             94            113         13         16           2   
51             75            131         18         20           1   
52            304            340          1          5           1   
53            241            243          1          3           2   
54             29            237          1         13           1   
55            289            316          1          4           1   
56            

In [15]:
# data_file = dataio.I3File(DATA_PATH)
# data_file.pop_frame()
# frame = data_file.pop_frame()
# print(frame["PMT_Response_nonoise"])

# 3.1: I3MCTree 

In [16]:
data_file = dataio.I3File(DATA_PATH)
data_file.pop_frame()
frame = data_file.pop_frame()
print(frame["I3MCTree"])

[I3MCTree:
  13 NuMu (46.7688m, 344.607m, 276.395m) (122.007deg, 79.0101deg) 0ns 119.301GeV 0m Primary
    7 MuMinus (46.7688m, 344.607m, 276.395m) (122.952deg, 80.7891deg) 0ns 97.2484GeV nanm MCTrack
    10 Hadrons (46.7688m, 344.607m, 276.395m) (117.605deg, 71.6458deg) 0ns 22.0522GeV nanm Cascade
]


# 3.2: I3MCTree_postprop

In [17]:
data_file = dataio.I3File(DATA_PATH)
data_file.pop_frame()
frame = data_file.pop_frame()
print(frame['I3MCTree_postprop'])

[I3MCTree:
  13 NuMu (46.7688m, 344.607m, 276.395m) (122.007deg, 79.0101deg) 0ns 119.301GeV 0m Primary
    7 MuMinus (46.7688m, 344.607m, 276.395m) (122.952deg, 80.7891deg) 0ns 97.2484GeV 377.572m Dark
      17 MuMinus (46.7688m, 344.607m, 276.395m) (123.011deg, 80.8016deg) 0ns 97.2484GeV 19.2187m Null
      18 DeltaE (44.1951m, 328.693m, 286.858m) (123.011deg, 80.8016deg) 64.1067ns 0.77235GeV 19.2187m Null
      19 MuMinus (44.1951m, 328.693m, 286.858m) (122.996deg, 80.7741deg) 64.1067ns 91.808GeV 12.7465m Null
      20 DeltaE (42.4816m, 318.14m, 293.799m) (122.996deg, 80.7741deg) 106.624ns 1.85901GeV 31.9652m Null
      21 MuMinus (42.4816m, 318.14m, 293.799m) (122.877deg, 80.754deg) 106.624ns 86.8552GeV 15.7771m Null
      22 DeltaE (40.3511m, 305.072m, 302.379m) (122.877deg, 80.754deg) 159.251ns 0.977276GeV 47.7423m Null
      23 MuMinus (40.3511m, 305.072m, 302.379m) (122.821deg, 80.9043deg) 159.251ns 82.0521GeV 93.6394m Null
      24 MuMinus (27.8353m, 227.424m, 353.196m) (122.68

# 3.3: MMCTrackList

In [18]:
data_file = dataio.I3File(DATA_PATH)
data_file.pop_frame()
frame = data_file.pop_frame()
print(frame['MMCTrackList'])

# 4.1: Noise_K40

In [19]:
nk40_table = FrameKeyToTable(
    data_path=DATA_PATH,
    frame_key="Noise_K40",     # I3Map<OMKey, vector<I3RecoPulse>>
)

df_nk40 = nk40_table.to_dataframe(max_events=300)
df_nk40

min_string_id  max_string_id  min_om_id  max_om_id  min_pmt_id  \
0               1            340          1         20           1   
1               1            340          1         20           1   
2               1            340          1         20           1   
3               1            340          1         20           1   
4               1            340          1         20           1   
5               1            340          1         20           1   
6               1            340          1         20           1   
7               1            340          1         20           1   
8               1            340          1         20           1   
9               1            340          1         20           1   
10              1            340          1         20           1   
11              1            340          1         20           1   
12              1            340          1         20           1   
13              1            340          1         20           1   
14              1            340          1         20           1   
15              1            340          1         20           1   
16              1            340          1         20           1   
17              1            340          1         20           1   
18              1            340          1         20           1   
19              1            340          1         20           1   
20              1            340          1         20           1   
21              1            340          1         20           1   
22              1            340          1         20           1   
23              1            340          1         20           1   
24              1            340          1         20           1   
25              1            340          1         20           1   
26              1            340          1         20           1   
27              1            340          1         20           1   
28              1            340          1         20           1   
29              1            340          1         20           1   
30              1            340          1         20           1   
31              1            340          1         20           1   
32              1            340          1         20           1   
33              1            340          1         20           1   
34              1            340          1         20           1   
35              1            340          1         20           1   
36              1            340          1         20           1   
37              1            340          1         20           1   
38              1            340          1         20           1   
39              1            340          1         20           1   
40              1            340          1         20           1   
41              1            340          1         20           1   
42              1            340          1         20           1   
43              1            340          1         20           1   
44              1            340          1         20           1   
45              1            340          1         20           1   
46              1            340          1         20           1   
47              1            340          1         20           1   
48              1            340          1         20           1   
49              1            340          1         20           1   
50              1            340          1         20           1   
51              1            340          1         20           1   
52              1            340          1         20           1   
53              1            340          1         20           1   
54              1            340          1         20           1   
55              1            340          1         20           1   
56            

In [20]:
# data_file = dataio.I3File(DATA_PATH)
# data_file.pop_frame()
# frame = data_file.pop_frame()
# print(frame["Noise_K40"])

# 4.2: Noise_Dark

In [21]:
nd_table = FrameKeyToTable(
    data_path=DATA_PATH,
    frame_key="Noise_Dark",
)
df_nd = nd_table.to_dataframe(max_events=300)
df_nd

min_string_id  max_string_id  min_om_id  max_om_id  min_pmt_id  \
0               1            340          1         20           1   
1               1            340          1         20           1   
2               1            340          1         20           1   
3               1            340          1         20           1   
4               1            340          1         20           1   
5               1            340          1         20           1   
6               1            340          1         20           1   
7               1            340          1         20           1   
8               1            340          1         20           1   
9               1            340          1         20           1   
10              1            340          1         20           1   
11              1            340          1         20           1   
12              1            340          1         20           1   
13              1            340          1         20           1   
14              1            340          1         20           1   
15              1            340          1         20           1   
16              1            340          1         20           1   
17              1            340          1         20           1   
18              1            340          1         20           1   
19              1            340          1         20           1   
20              1            340          1         20           1   
21              1            340          1         20           1   
22              1            340          1         20           1   
23              1            340          1         20           1   
24              1            340          1         20           1   
25              1            340          1         20           1   
26              1            340          1         20           1   
27              1            340          1         20           1   
28              1            340          1         20           1   
29              1            340          1         20           1   
30              1            340          1         20           1   
31              1            340          1         20           1   
32              1            340          1         20           1   
33              1            340          1         20           1   
34              1            340          1         20           1   
35              1            340          1         20           1   
36              1            340          1         20           1   
37              1            340          1         20           1   
38              1            340          1         20           1   
39              1            340          1         20           1   
40              1            340          1         20           1   
41              1            340          1         20           1   
42              1            340          1         20           1   
43              1            340          1         20           1   
44              1            340          1         20           1   
45              1            340          1         20           1   
46              1            340          1         20           1   
47              1            340          1         20           1   
48              1            340          1         20           1   
49              1            340          1         20           1   
50              1            340          1         20           1   
51              1            340          1         20           1   
52              1            340          1         20           1   
53              1            340          1         20           1   
54              1            340          1         20           1   
55              1            340          1         20           1   
56            

In [22]:
# data_file = dataio.I3File(DATA_PATH)
# data_file.pop_frame()
# frame = data_file.pop_frame()
# print(frame["Noise_Dark"])

# 5.1: I3MCTree_RNGState

In [23]:
data_file = dataio.I3File(DATA_PATH)
data_file.pop_frame()
frame = data_file.pop_frame()
frame['I3MCTree_RNGState']

In [24]:
# hof

In [25]:
data_file = dataio.I3File(DATA_PATH)
data_file.pop_frame()
frame = data_file.pop_frame()
print(frame['TriggerTime_3PMT_1DOM'])

I3Double(458)


In [26]:
frame['TriggerTime_3PMT_1DOM'].value

458.0